crates/red_knot_python_semantic/resources/mdtest/attributes.md

```diff
@@ -806,6 +809,9 @@ def _(flag: bool, flag1: bool, flag2: bool):
 
     # error: [possibly-unbound-attribute] "Attribute `x` on type `Literal[C1, C2, C3]` is possibly unbound"
     reveal_type(C.x)  # revealed: Unknown | Literal[1, 2, 3]
+
+    # error: [possibly-unbound-attribute] "Attribute `x` on type `C1 | C2 | C3` is possibly unbound"
+    reveal_type(C().x)  # revealed: Unknown | Literal[1, 2, 3]
```

Comment:
I guess for this one, I'm not totally clear on how it's different to https://github.com/astral-sh/ruff/pull/16363/files#r1969441733. All members of the union have an `x` attribute here, and like https://github.com/astral-sh/ruff/pull/16363/files#r1969441733, it's still an instance attribute that we can't verify has been definitely defined. Like https://github.com/astral-sh/ruff/pull/16363/files#r1969441733, there's still the issue that it's (unfortunately) not uncommon in Python to initialize attributes on instances after the instance has been constructed.

Reply:
That's a good question. I modeled it after the behavior on the class object, but we may want to design that differently. Should possible unboundness always be ignored for instance members?

Reply:
Oops. Didn't see @carljm's answer. Does that mean that we can keep the current behavior for now?

Reply:
> Should possible unboundness always be ignored for instance members?

I would argue yes, but I'm also fine with deferring this for now. Could possibly add a TODO saying that this could be revisited in the future?

Reply:
Done.